### ISSUE TITLE
Cannot disable idea.auto.reload.plugins

### ISSUE THREAD

**Issue by gharrma** [JetBrains/gradle-intellij-plugin#476]
The gradle-intellij-plugin sets the `idea.auto.reload.plugins` system property by default. It seems I cannot override it.

I tried:
```gradle
runIde {
    systemProperty('idea.auto.reload.plugins', false) // also tried 'null'
}
```
It does not work because `DynamicPluginVfsListener` (in IDEA) only checks whether the property is defined---it does not bother checking what its value is.

I also tried:
```gradle
runIde {
    systemProperties.remove('idea.auto.reload.plugins')
}
```
It does not work because this code runs before `RunIdeBase.configureSystemProperties()`, where the property is added.

**Comment by zolotov** [comment]
Hi,

> It does not work because DynamicPluginVfsListener (in IDEA) only checks whether the property is defined---it does not bother checking what its value is.

I think it should be changed in IDEA. cc @yole 

Why do you need to disable it in the first place?

**Comment by gharrma** [comment]
> Why do you need to disable it in the first place?

My plugin does not yet unload successfully, so the auto-reload features sometimes breaks my plugin while I'm trying to run it.

(Auto-reload seems to trigger sometimes even when I have not made any changes to my plugin nor rebuilt it. I don't know if that's a bug in `DynamicPluginVfsListener`, or if I'm just doing something wrong.)

I just realized that I can disable unloading by overriding `DynamicPluginListener.checkUnloadPlugin()`, so I will do that for now.

p.s. I'm not sure whether my plugin will ever be able to be unloaded, because it includes a Java instrumentation agent.

**Comment by Almighty-Alpaca** [comment]
I currently face the same issue. After some time IntelliJ randomly tries to reload my plugin. In my case this fails because my plugin loads a native library that cannot be unloaded.

@gharrma I admit it's not a great solution but as a workaround I currently use this
```kotlin
class MyDynamicPluginListener : DynamicPluginListener {
    override fun checkUnloadPlugin(pluginDescriptor: IdeaPluginDescriptor) {
        if (pluginDescriptor.pluginId == PluginId.getId("my.plugin.id")) {
            throw Class
                    .forName("com.intellij.ide.plugins.CannotUnloadPluginException")
                    .constructors
                    .first()
                    .newInstance("unsupported") as Exception
        }
    }
}

```

**Comment by zolotov** [comment]
> It does not work because DynamicPluginVfsListener (in IDEA) only checks whether the property is defined---it does not bother checking what its value is.

this will be fixed in one of 2020.1 updates, likely in 2020.1.2
